src/btree/bt_sync.c

```diff
@@ -297,8 +209,133 @@ __sync_delete_obsolete_ref(WT_SESSION_IMPL *session, WT_REF *ref)
       (void *)ref, tag, obsolete ? "" : "not ",
```

Comment:
Simply evict the clean page can get it removed?

Reply:
No, it doesn't clean automatically if we evict. There are two ways to clean an obsolete in-memory page, 

1. We need to either mark it dirty so that the next reconciliation will write an empty page during the reconciliation.
2. Evict the page and free it in the next checkpoint. 

Currently, the second option is implemented to free the entire page instead of freeing the page key by key. During the next checkpoint, the obsolete page is on-disk. So we mark it as deleted with no `page_del` information and whenever the internal page is reconciled, it will be freed.

Reply:
Is it possible that this page is read into memory again is someone searching a key on it but return not found? If it read into memory again, then it is not cleaned in the next checkpoint and we need to wait for the next and next checkpoint I guess.

Reply:
Yes, your doubt is correct. Currently, we have a logic to skip reading any leaf pages into memory if the aggregated stop timestamp of the leaf page is visible to the reader.